components/governance/org.wso2.carbon.governance.taxonomy/src/main/java/org/wso2/carbon/governance/taxonomy/services/TaxonomyServicesImpl.java

```diff
@@ -25,92 +27,125 @@
 import org.wso2.carbon.governance.taxonomy.util.CommonUtils;
 import org.wso2.carbon.registry.common.services.RegistryAbstractAdmin;
 import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.user.api.UserStoreException;
 import org.xml.sax.SAXException;
 import java.io.IOException;
 import javax.xml.parsers.ParserConfigurationException;
 import javax.xml.xpath.XPathExpressionException;
 
 public class TaxonomyServicesImpl extends RegistryAbstractAdmin implements ITaxonomyServices {
-
+    private static final Log log = LogFactory.getLog(TaxonomyServicesImpl.class);
     /**
      * This method will add user defined taxonomy into taxonomy.xml
      *
      * @param payload String
      * @return boolean
-     * @throws RegistryException
+     * @throws Exception
      */
     @Override public boolean
-    addTaxonomy(String payload) throws RegistryException {
+    addTaxonomy(String payload) throws Exception {
         TaxonomyManager taxonomyManager = new TaxonomyManager();
         try {
             TaxonomyBean documentBean = CommonUtils.documentBeanBuilder(payload);
             return taxonomyManager.addTaxonomy(documentBean);
         } catch (ParserConfigurationException e) {
-            throw new RegistryException("Error occurred while parsing payload ", e);
+            log.error("Error occurred while parsing payload ", e);
+            throw (e);
         } catch (IOException e) {
-            throw new RegistryException("Error occurred while building document ", e);
+            log.error("Error occurred while building document ", e);
+            throw (e);
         } catch (SAXException e) {
-            throw new RegistryException("Error occurred while creating document instance ", e);
+            log.error("Error occurred while creating document instance ", e);
+            throw (e);
```

Comment:
We can't catch specific type exceptions inside jsp since we can't send specific type exception through stub,.  So we have to catch and log the specific type exception before sending to stub.